case-lib/hijack.sh

```diff
@@ -48,9 +48,17 @@ function exit()
         done
     fi
 
-    # check if function already defined
+    # check if function already defined, this should be always true.
     # on exit check whether pulseaudio is disabled
-    [[ $(declare -f func_lib_restore_pulseaudio) ]] && func_lib_restore_pulseaudio
+    if [[ $(declare -f func_lib_restore_pulseaudio) ]]; then
+        func_lib_restore_pulseaudio
+        ret=$?
+    fi
+    # if failed to restore pulseaudio, even test caes passed, set exit status to ret
+    # to make test case failed. this helps to dectect pulseaudio failures.
+    if [ $exit_status -eq 0 -a $ret -ne 0 ]; then
```

Comment:
if `$ret` is not setup, here will cause a error

Reply:
I thought [[ $(declare -f func_lib_restore_pulseaudio) ]] will always return true, seems I am wrong